addons/web_editor/controllers/main.py

```diff
@@ -309,14 +309,14 @@ def add_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=Fa
         if is_image:
             format_error_msg = _("Uploaded image's format is not supported. Try with: %s", ', '.join(SUPPORTED_IMAGE_EXTENSIONS))
             try:
-                data = tools.image_process(data, size=(width, height), quality=quality, verify_resolution=True)
                 mimetype = guess_mimetype(data)
                 if mimetype not in SUPPORTED_IMAGE_MIMETYPES:
                     return {'error': format_error_msg}
-            except UserError:
+                data = tools.image_process(data, size=(width, height), quality=quality, verify_resolution=True)
+            except UserError as e:
```

Comment:
```suggestion
            except (ValueError, UserError) as e:
```
And move comment of UserError inside this by adapting statement like "When UserError thrown consider an image..."

Reply:
Nice insight!
pushed changes. Thanks!